lib/scanner/walk.go

```diff
@@ -559,3 +562,54 @@ type noCurrentFiler struct{}
 func (noCurrentFiler) CurrentFile(name string) (protocol.FileInfo, bool) {
 	return protocol.FileInfo{}, false
 }
+
+// ScannerLimiter should limit parallel folder scanning
+type ScannerLimiter interface {
+	Aquire(ctx context.Context, d ...string)
+	Release(d ...string)
+}
+
+func NewScannerLimiter(single bool) ScannerLimiter {
+	if single {
+		l.Infoln("single global folderScanner limit ")
+		return &singleGlobalScannerLimiter{
+			sem: semaphore.New(1),
```

Comment:
This whole thing is equivalent to a regular `sync.Mutex` or a single-buffered `chan struct{}`. I do not think we should pull in that dependency. It's also racy and not great (I happened to glance at it) so I think we should just use a channel if we need to do more than `1` here.